src/explore-education-statistics-admin/src/pages/release/data/components/DataFileUploadForm.tsx

```diff
@@ -114,15 +121,56 @@ function baseErrorMappings(
 interface Props {
   dataFiles?: DataFile[];
   isDataReplacement?: boolean;
+  releaseId?: string;
   onSubmit: (values: DataFileUploadFormValues) => void | Promise<void>;
 }
 
 export default function DataFileUploadForm({
   dataFiles,
   isDataReplacement = false,
+  releaseId,
   onSubmit,
 }: Props) {
   const [selectedFileType, setSelectedFileType] = useState<FileType>('csv');
+  const [showModal, toggleShowModal] = useToggle(false);
+  const [bulkUploadPlan, setBulkUploadPlan] = useState<ArchiveDataSetFile[]>();
+
+  async function getBulkUploadPlan(zipFile: File | null | undefined) {
```

Comment:
I think we added the union type here because `DataFileUploadFormValues.bulkZipFile` is optional, and possibly because a file input can be null/undefined.
![image](https://github.com/user-attachments/assets/ae0bf633-a1b2-449c-94a4-d9f468362995)

Reply:
Happy to go with your preference - I think we're splitting hairs on where the `if` goes..
I'm suggesting something like
```ts
if (order) getMyFood(order) // getMyfood(order: Order)
```
in comparison to the implementation
```ts
getMyFood(order: Order | undefined) {
  if(!order) return // do nothing
  // ...
}
```